src/node_modules.cc

```diff
@@ -0,0 +1,442 @@
+      CHECK(file_url);
+      auto file_path = url::FileURLToPath(realm->env(), *file_url);
+      CHECK(file_path.has_value());
+      THROW_ERR_INVALID_PACKAGE_CONFIG(
+          realm->isolate(),
+          "Invalid package config %s while importing \"%s\" from %s.",
+          path.data(),
+          error_context->specifier.c_str(),
+          file_path->c_str());
+    } else {
+      THROW_ERR_INVALID_PACKAGE_CONFIG(
+          realm->isolate(), "Invalid package config %s.", path.data());
+    }
+
+    return std::nullopt;
+  };
+
+  if (error || document.get_object().get(main_object)) {
+    return throw_invalid_package_config();
+  }
+
+  simdjson::ondemand::raw_json_string key;
+  simdjson::ondemand::value value;
+  PackageConfig package_config{};
+
+  for (auto field : main_object) {
+    // Throw error if getting key or value fails.
+    if (field.key().get(key) || field.value().get(value)) {
+      return throw_invalid_package_config();
+    }
+
+    if (key == "name") {
+      std::string_view name;
+      if (!value.get_string().get(name)) {
+        package_config.name = std::string(name);
+      }
+    } else if (key == "main") {
+      std::string_view main;
+      if (!value.get_string().get(main)) {
+        package_config.main = std::string(main);
+      }
+    } else if (key == "exports") {
+      simdjson::ondemand::json_type exports_type;
+      if (!value.type().get(exports_type)) {
+        std::string_view exports;
+        switch (exports_type) {
+          case simdjson::ondemand::json_type::object: {
+            simdjson::ondemand::object exports_object;
+            if (!value.get_object().get(exports_object) &&
+                !exports_object.raw_json().get(exports)) {
+              package_config.exports = std::string(exports);
+            }
+            break;
+          }
+          case simdjson::ondemand::json_type::array: {
+            simdjson::ondemand::array exports_array;
+            if (!value.get_array().get(exports_array) &&
+                !exports_array.raw_json().get(exports)) {
+              package_config.exports = std::string(exports);
+            }
+            break;
+          }
+          case simdjson::ondemand::json_type::string: {
+            if (!value.get_string().get(exports)) {
+              package_config.exports = std::string(exports);
+            }
+            break;
+          }
+          default:
+            break;
+        }
+      }
+    } else if (key == "imports") {
+      simdjson::ondemand::json_type imports_type;
+      if (!value.type().get(imports_type)) {
+        std::string_view imports;
+        switch (imports_type) {
+          case simdjson::ondemand::json_type::object: {
+            simdjson::ondemand::object imports_object;
+            if (!value.get_object().get(imports_object) &&
+                !imports_object.raw_json().get(imports)) {
+              package_config.imports = std::string(imports);
+            }
```

Comment:
if(value.get_array().get(import_object)) {
  // the document is likely busted, error.
}
if(std::string_view import_object; import_object.raw_json().get(import_object)) {
  // almost surely a fatal error: the JSON document is busted.
} else {
  package_config.imports = std::string(import_object);
}